fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -413,8 +413,10 @@ public void close() {
                 } else {
                     int currTries = tries.getAndIncrement();
                     FDBException fdbE = getFDBException(e);
-                    if (currTries < config.maxRetries && fdbE != null && lessenWorkCodes.contains(fdbE.getCode())) {
-                        if (handleLessenWork != null) {
+                    if (currTries < config.maxRetries && (
+                            (fdbE != null && lessenWorkCodes.contains(fdbE.getCode())) ||
+                            (e instanceof RecordStoreStaleMetaDataVersionException))) {
```

Comment:
I suppose it's worth noting that this is only the right thing to do if given a `RecordMetaDataProvider` that knows to refresh (somehow) on this error, or this will just spin, hitting the same error each time. I also don't know if the `RecordStoreStaleMetaDataVersionException` will be wrapped in a `CompletionException` at this point. I don't...think so, but I'm not certain.

Reply:
Yeah. It sounds better to retry on the developer side to start a new `OnlineIndexer` instead. Closing this PR.